cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudEfsAttributes.java

```diff
@@ -0,0 +1,241 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.sequenceiq.cloudbreak.cloud.model.filesystem.efs.LifeCycleState;
+
+@JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CloudEfsAttributes {
+    public static final String EFS_TAGKEY_NAME = "Name";
+
+    public static final String EFS_TAGKEY_NAME_VALUE_DEFAULT = "defaultName";
+
+    private String fileSystemId;
+
+    private String creationToken;
+
+    private Boolean deleteOnTermination;
+
+    private Map<String, String> tags;
+
+    private String performanceMode;
+
+    private String throughputMode;
+
+    private Double provisionedThroughputInMibps;
+
+    private Boolean encrypted;
+
+    private String kmsKeyId;
+
+    private LifeCycleState fileState;
+
+    @JsonCreator
+    public CloudEfsAttributes(@JsonProperty("creationToken") String creationToken, @JsonProperty("deleteOnTermination") Boolean deleteOnTermination,
+            @JsonProperty("tags") Map<String, String> tags, @JsonProperty("performanceMode") String performanceMode,
+            @JsonProperty("throughputMode") String throughputMode, @JsonProperty("provisionedThroughputInMibps") Double provisionedThroughputInMibps,
+            @JsonProperty("encrypted") Boolean encrypted, @JsonProperty("kmsKeyId") String kmsKeyId) {
+        this.creationToken = creationToken;
+        this.deleteOnTermination = deleteOnTermination;
+        this.tags = tags;
+        this.performanceMode = performanceMode;
+        this.throughputMode = throughputMode;
+        this.provisionedThroughputInMibps = provisionedThroughputInMibps;
+        this.encrypted = encrypted;
+        this.kmsKeyId = kmsKeyId;
+
+        // we don't know the value untile the response from AWS
+        this.fileSystemId = null;
+        this.fileState = LifeCycleState.PREPARE;
+    }
+
+    public CloudEfsAttributes(CloudEfsAttributes oriAttributes) {
+        this(oriAttributes.creationToken, oriAttributes.deleteOnTermination, new HashMap<String, String>(oriAttributes.tags),
+                oriAttributes.performanceMode, oriAttributes.throughputMode, oriAttributes.provisionedThroughputInMibps, oriAttributes.encrypted,
+                oriAttributes.kmsKeyId);
+        setFileSystemId(oriAttributes.fileSystemId);
+        setFileState(oriAttributes.fileState);
+    }
+
+    public String getName() {
+        if (tags == null) {
+            return EFS_TAGKEY_NAME_VALUE_DEFAULT;
+        }
+
+        return tags.getOrDefault(EFS_TAGKEY_NAME, EFS_TAGKEY_NAME_VALUE_DEFAULT);
+    }
+
+    public void setName(String efsName) {
+        if (tags == null) {
```

Comment:
This is not thread-safe, could you please init at construct time?

Reply:
fixed